src/DataCollector/RequestDataCollector.php

```diff
@@ -33,7 +33,9 @@ public function collect(Request $request, Response $response, \Exception $except
      */
     public function getRequestCount(): int
     {
-        return count($this->data['requests']);
+        return array_sum(array_map(function (array $requests) {
+            return count($requests);
+        }, $this->data['requests']));
```

Comment:
Since there is a method to return the requests `$this->getRequests(): array` you should use it in here.
And I think you should add a fallback for `getRequests()` if index `'requests'` is not set because the method is public and could be invoked at a time where `collect()` wasn't invoked already.